server/backends/blobstore/gcs/gcs.go

```diff
@@ -134,3 +136,24 @@ func (g *GCSBlobStore) BlobExists(ctx context.Context, blobName string) (bool, e
 		return false, err
 	}
 }
+
+func (g *GCSBlobStore) Writer(ctx context.Context, blobName string) (interfaces.CommittedWriteCloser, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	bw := g.bucketHandle.Object(util.BlobPath(blobName)).NewWriter(ctx)
+
+	zw := util.NewCompressWriter(bw)
+	cwc := ioutil.NewCustomCommitWriteCloser(zw)
+	cwc.CommitFn = func(int64) error {
+		if compresserCloseErr := zw.Close(); compresserCloseErr != nil {
+			cancel()       // Don't try to finish the commit op if Close() failed.
+			_ = bw.Close() // Canceling the context makes any error here meaningless.
```

Comment:
It's not 100% clear from the GCS client docs, but cancelling the context automatically closes the writer so it is likely safer not to try to call `Close` as well since there might be a race between the context cancellation and the close that could lead to the write succeeding.

Reply:
Thanks, didn't realize; fixed!